src/compose/ComposeBox.js

```diff
@@ -320,14 +322,18 @@ class ComposeBox extends PureComponent<Props, State> {
   };
 
   handleSend = () => {
-    const { dispatch, narrow, caughtUp, _ } = this.props;
+    const { dispatch, narrow, caughtUp, _, isTopicMandatory } = this.props;
     const { message } = this.state;
+    const topic = this.state.topic;
 
     if (!caughtUp.newer) {
       showErrorAlert(_('Failed to send message'));
       return;
     }
-
+    if (isTopicMandatory && (topic === '(no topic)' || !topic)) {
```

Comment:
It looks like the web app only does the empty-string check; it doesn't check for '(no topic)': https://github.com/zulip/zulip/blob/master/static/js/compose.js#L732

Let's do that, to match the web app's behavior but also to keep low the number of places where we use the string literal '(no topic)'. Ideally, I think, that string would be stored in a constant, with a comment pointing to a corresponding constant in zulip/zulip.git.